### DISCUSSION TITLE
Swiper as a 'popup' like Photoswipe

### DISCUSSION THREAD

**Discussion by inttyl** [nolimits4web/swiper#5698] [General]
Hi,

Is it possible to open Swiper in a overlay like the library photoswipe ( [Photoswipe](https://photoswipe.com) ) when we click into as image ?
I am trying to integrate swiper in a angular but it seems that  the images in swipee cannot be viewed  in a overlay like photoswipe.

Thanks for your help

**Comment by Arhell** [comment]
https://codepen.io/ezra_siton/pen/XNpJaX

**Reply by pmorch** [reply]
See also #4336

As far as i can see, https://codepen.io/ezra_siton/pen/XNpJaX does something different:

It uses [swiper on the "main page"](https://github.com/user-attachments/assets/be796934-379c-4c5a-9657-45b9cdd3daee), and then when you click on an image, it opens [Photoswipe in a lightbox](https://github.com/user-attachments/assets/507ae24c-a485-4192-b273-404a9c074b44).

What the OP is asking for is how to open **Swiper** in a overlay. https://codepen.io/ezra_siton/pen/XNpJaX opens **Photoswipe** in an overlay. Not the same thing at all.

**Comment by pmorch** [comment]
I just created https://codesandbox.io/p/sandbox/2hgrlw that does this as a proof of concept.

It is functional and shows how one can use the native `<dialog>` HTML element to achieve this, but a lot of styling is still required to get this to look great. Compare my  sandbox screenshot:

![my sandbox screenshot](https://github.com/user-attachments/assets/08210d7d-e18a-408b-bf14-170f479d961c)

to Photoswipe in a lightbox:

![Photoswipe in a lightbox](https://github.com/user-attachments/assets/507ae24c-a485-4192-b273-404a9c074b44).

So this is a proof-of-concept - nothing more.
